### ISSUE TITLE
Support for query parameterization

### ISSUE THREAD

**Issue by dylanscott** [googleapis/python-bigquery-dataframes#384]
I was wondering if there were plans to expose the ability to pass bound parameters (either named or qmark style) when constructing constructing a bigframes DataFrame e.g. via `Session.read_gbq_query`? My understanding is they would need to be thread through to the [`queryParameters` field](https://cloud.google.com/bigquery/docs/reference/rest/v2/Job#JobConfigurationQuery.FIELDS.query_parameters) on the `QueryJobConfig` that gets constructed.

I realize it's probably a bit of a niche use-case since this is a code-based API but for context we were looking to integrate support for bigframes in a product that allows users to build Python notebook-based interactive apps which may be used by external/untrusted users (not the ones writing the app/code) and so our SQL functionality relies on using parameterization whenever these kinds of inputs are involved.

**Comment by tswast** [comment]
In pandas-gbq this is exposed via a generic "configuration" parameter, where the JobConfig REST object can be passed directly as a dictionary.

I suggest that `read_gbq` and `read_gbq_query` add the same `configuration` argument in BigQuery DataFrames.

**Comment by tswast** [comment]
Draft PR: https://github.com/googleapis/python-bigquery-dataframes/pull/401

**Comment by tswast** [comment]
Expected usage will be the same as pandas-gbq. See: https://stackoverflow.com/a/62171130/101923

```python
query_config = {
    'query': {
        'parameterMode': 'NAMED',
        'queryParameters': [
            {
                'name': 'state',
                'parameterType': {'type': 'STRING'},
                'parameterValue': {'value': 'TX'}
            },
        ]
    }
}
df = pandas_gbq.read_gbq(sql, configuration=query_config)
```

**Comment by tswast** [comment]
This has been released in version 1.0.0
